vm/vm.go

```diff
@@ -22,8 +23,25 @@ func Run(program *Program, env interface{}) (out interface{}, err error) {
 		}
 	}()
 
-	out = vm.Run(program, env)
-	return
+	ch := make(chan interface{}, 1)
+	defer close(ch)
+
+	go func() {
+		ch <- vm.Run(program, env)
+	}()
+
+	timer := time.NewTimer(program.Timeout)
+	defer timer.Stop()
+
+	select {
+	case out = <-ch:
+		return
+	case <-timer.C:
+		// TODO: type error so it can be checked externally
+		err = fmt.Errorf("operation timedout. Current Timeout %v", program.Timeout)
+		return
```

Comment:
But this won't stop goroutine and code will be still working. This kind of early return can be done outsize Expr library same way.

Reply:
Thats a good point, the timeout would likely need to be paired with a `interrupt` channel to properly halt the parser. 

Something like [this](https://github.com/robertkrimen/otto/blob/15f95af6e78dcd2030d8195a138bd88d4f403546/cmpl_evaluate_expression.go#L16 ) at the [root](https://github.com/antonmedv/expr/blob/73253fa3d7f2902c5c3be75a9775b6f1aa47157e/compiler/compiler.go#L115) of the parser.